src/shared/pcre2-util.h

```diff
@@ -3,21 +3,20 @@
 
 #include "macro.h"
 
-#if HAVE_PCRE2
+typedef void pcre2_pattern;
 
-#define PCRE2_CODE_UNIT_WIDTH 8
-#include <pcre2.h>
+typedef enum {
+        PATTERN_COMPILE_CASE_AUTO,
+        PATTERN_COMPILE_CASE_SENSITIVE,
+        PATTERN_COMPILE_CASE_INSENSITIVE,
+        _PATTERN_COMPILE_CASE_MAX,
+        _PATTERN_COMPILE_CASE_INVALID = -EINVAL,
+} PatternCompileCase;
 
-extern pcre2_match_data* (*sym_pcre2_match_data_create)(uint32_t, pcre2_general_context *);
-extern void (*sym_pcre2_match_data_free)(pcre2_match_data *);
-extern void (*sym_pcre2_code_free)(pcre2_code *);
-extern pcre2_code* (*sym_pcre2_compile)(PCRE2_SPTR, PCRE2_SIZE, uint32_t, int *, PCRE2_SIZE *, pcre2_compile_context *);
-extern int (*sym_pcre2_get_error_message)(int, PCRE2_UCHAR *, PCRE2_SIZE);
-extern int (*sym_pcre2_match)(const pcre2_code *, PCRE2_SPTR, PCRE2_SIZE, PCRE2_SIZE, uint32_t, pcre2_match_data *, pcre2_match_context *);
-extern PCRE2_SIZE* (*sym_pcre2_get_ovector_pointer)(pcre2_match_data *);
+int pcre2_pattern_compile(const char *pattern, PatternCompileCase case_, pcre2_pattern **ret);
+int pcre2_pattern_matches(pcre2_pattern *compiled_pattern, const char *message, size_t size, size_t *ret_ovec);
+void *pcre2_pattern_free(pcre2_pattern *p);
```

Comment:
we are kinda invading pcre2's own function naming namespace if we name these helper calls pcre2_xyz.

we do something like at variou splaces, but maybe we shouldn't...

Maybe drop the pcre2_ prefix, and just use pattern_ as prefix? after all the enum above you already callsed Pattern… hence this would match?